### PR TITLE
Fix Bug Causing Editor To Continue Server Connection Attempts After Exiting Play Mode

### LINKED ISSUE #10749: When the Editor Server crashes, the Editor can get stuck in the infinite loop of retries
**Describe the bug**
The Editor becomes pretty much unresponsive if the editor-server crashes during connection step 2.

image

I advise that we at least limit the number of retries and then bail out back to the Edit mode.

**Steps to reproduce**
1. Set up a multiplayer project
2. Enter game mode

**Expected behavior**
If the server crashes during connection, the Editor should NOT be re-trying forever. 

**Actual behavior**
The Editor is stuck trying forever even after pressing ESC.

**Screenshots/Video**
image

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows

### PR DESCRIPTION
**Pull request by AMZN-Gene**
## What does this PR do?
Makes sure that Multiplayer Editor Server connection attempts stop when exiting game mode
Adding max connection attempts editorsv_max_connection_attempts

Fixes #10749

## How was this PR tested?
Ran the editor multiplayer with debug code that causes the connection to faill and see the max attempts are reached and message is displayed on screen.

### CODE REVIEW

**Review by AMZN-Olex — APPROVED** [review]

**Review by AMZN-puvvadar — APPROVED** [review]

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp [hunk 1]
```diff
@@ -186,12 +186,17 @@ namespace Multiplayer
 
     void MultiplayerConnectionViewportMessageSystemComponent::OnEditorConnectionAttempt(uint16_t connectionAttempts)
     {
-        m_centerViewportDebugText = AZStd::fixed_string<128>::format(OnEditorConnectionAttemptMessage, connectionAttempts);
+        m_centerViewportDebugText = AZStd::fixed_string<MaxMessageLength>::format(OnEditorConnectionAttemptMessage, connectionAttempts);
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::OnAllConnectionAttemptsFailed()
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp:192]: nit: OnEditorConnectionFailed ? to match OnEditorConnectionAttempt

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.h [hunk 2]
```diff
@@ -24,11 +24,13 @@ namespace Multiplayer
         , MultiplayerEditorServerNotificationBus::Handler
     {
     public:
+        static constexpr int MaxMessageLength = 256;
         static constexpr char CenterViewportDebugTitle[] = "Multiplayer Editor";
         static constexpr char ClientStatusTitle[] = "Multiplayer Client Status:";
         static constexpr char OnServerLaunchedMessage[] = "(1/3) Launching server...";
         static constexpr char OnServerLaunchFailMessage[] = "(1/3) Could not launch editor server.\nSee console for more info.";
-        static constexpr char OnEditorConnectionAttemptMessage[] = "(2/3) Editor tcp connection attempt #%i.";
+        static constexpr char OnEditorConnectionAttemptMessage[] = "(2/3) Editor TCP connection attempt #%i.";
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.h:32]: I would love to see this as more. "Attempting to connect to server via TCP. Attempt x of x" - as is IMHO it doesn't clearly state whats going on.

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.h [hunk 3]
```diff
@@ -24,11 +24,13 @@ namespace Multiplayer
...
+        static constexpr char OnAllConnectionAttemptsFailedMessage[] = "(2/3) All connection attempts failed!\nConnection limit is defined by 'editorsv_max_connection_attempts'\nPlease exit play mode and try again.";
```
Comment (lmbr-pip) [inline Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.h:33]: I would keep this simpler - "Failed to connect to server after x attempts. Please exist play mode and try again"

Changing the `editorsv_max_connection_attempts` will not really help here in most cases AFAIK so would leave that to docs as an "advanced setting"

**Review by lmbr-pip — COMMENTED** [review]
LG - nice update to the behaviour.

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp (hunk 1 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp]: <3 fixed!

File: Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.h (hunk 3 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.h]: Perfect. Fixed :)

**Pull request merged**